commit/plugin.go

```diff
@@ -29,8 +40,10 @@ type Plugin struct {
 	lggr              logger.Logger
 	homeChain         reader.HomeChain
 	reportingCfg      ocr3types.ReportingPluginConfig
-	chainSupport      ChainSupport
-	observer          Observer
+	chainSupport      shared.ChainSupport
+	merkleProcessor   shared.PluginProcessor[merkleroot.Query, merkleroot.Observation, merkleroot.Outcome]
+	tokenProcessor    shared.PluginProcessor[tokenprice.Query, tokenprice.Observation, tokenprice.Outcome]
+	feeProcessor      shared.PluginProcessor[chainfee.Query, chainfee.Observation, chainfee.Outcome]
```

Comment:
You're not using these generic types so `shared.PluginProcessor` should define them as `any` for a simpler interface.

If you don't need the types, you can now use the generic processor:
```suggestion
	merkleProcessor   shared.PluginProcessor
	tokenProcessor    shared.PluginProcessor
	feeProcessor      shared.PluginProcessor
```

If you do need the types in some cases, specify the concrete type:
```suggestion
	merkleProcessor   merkleroot.Processor // because we need to call ObserveFChain()'
	tokenProcessor    shared.PluginProcessor
	feeProcessor      shared.PluginProcessor
```

Reply:
I think we'll need the types when we start testing. and using the plugin makes the one mock enough to use. WDYT?

Reply:
We've discussed on a call and decided to keep as is for testability and we'll revisit this if we can make it more generic later.